src/app/landing/shared/services/repository.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable, OnDestroy } from "@angular/core";
+import { CoursesByTechnology, CoursesRepository } from "../interfaces/course";
+import { mockCourses } from "./mockCourses";
+import { Observable, Subject, delay, of, retry, shareReplay, takeUntil } from "rxjs";
+import { mockStutendsFeedback } from "./mockStudentsFeedback";
+import { StudentFeedback, StudentFeedbacksRepository } from "../interfaces/student-feedback";
+import { Mentor, MentorsRepository } from "../interfaces/mentor";
+import { mockMentors } from "./mockMentors";
+
+@Injectable({providedIn: 'root'})
+export class RepositoryService implements 
+    CoursesRepository, 
+    StudentFeedbacksRepository, 
+    MentorsRepository,
+    OnDestroy
+{
+    private coursesCache: Observable<CoursesByTechnology> | null = null
+    private studentsFeedbackCache: Observable<StudentFeedback[]> | null = null
+    private mentorsCache: Observable<Mentor[]> | null = null
+
+    private unsubscribe$ = new Subject()
+
+    private prepareAPIRequest<T>(request: Observable<T>): Observable<T>{
+        return request.pipe(
+            retry({delay: 1000}),
+            takeUntil(this.unsubscribe$),
```

Comment:
Why takeUntil with unsubscribe is in service? unsubscribe should be in component.

In Angular V16 we can use takeUntilDestroyed operator. Consider maybe to use is on your code